src/openforms/variables/models.py

```diff
@@ -126,12 +127,25 @@ def request_arguments(self, context: DataMapping) -> dict:
         #
         # extra knowledge not in the RFC: latin1 is a different name for ISO-8859-1
 
+        # Explicitly cast leaf values to strings to avoid localization
+        StringLeaved = str | list["StringLeaved"] | dict[str, "StringLeaved"]
+
+        def cast_leaves(thing: any, cast: Callable[[any], str] = str) -> StringLeaved:
+            match thing:
+                case [*xs]:
+                    return [cast_leaves(x, cast) for x in xs]
+                case {**d}:
+                    return {k: cast_leaves(v, cast) for k, v in d.items()}
+                case _:
+                    return cast(thing)
```

Comment:
this is essentially the same as `openforms.formio.variables._recursive_apply` / `openforms.formio.variables.render`, so we're ending up with different flavours of the same thing.

This is also very much operating on the "variables/formio space", so perhaps this should become a proper utility function  in a centralized location in the codebase.